lib/dotcom_web/components/mode_icons.ex

```diff
@@ -0,0 +1,112 @@
+defmodule DotcomWeb.Components.ModeIcons do
+  import DotcomWeb.Components.Badges
+  import DotcomWeb.Components.Svg
+  use Phoenix.Component
+
+  attr :route_name, :string, required: true
+
+  attr :type, :atom,
+    values: [
+      :subway,
+      :commuter_rail,
+      :bus,
+      :ferry,
+      :logan_express,
+      :massport_shuttle,
+      :orange_line,
+      :red_line,
+      :green_line,
+      :blue_line,
+      :mattapan_line
+    ],
+    required: true
+
+  attr :class, :string, default: ""
+
+  def mode_icon(%{type: :bus} = assigns) do
+    ~H"""
+    <.rect_icon class={"bg-yellow-500 #{@class}"}><%= @route_name %></.rect_icon>
+    """
+  end
+
+  def mode_icon(%{type: :silver_line} = assigns) do
+    ~H"""
+    <.rect_icon class={"bg-silver-500 text-white #{@class}"}><%= @route_name %></.rect_icon>
+    """
+  end
+
+  def mode_icon(%{type: :massport_shuttle} = assigns) do
+    ~H"""
+    <.circle_icon class={"bg-navy-500 text-md #{@class}"}><%= @route_name %></.circle_icon>
+    """
+  end
+
+  def mode_icon(%{type: :blue_line} = assigns) do
+    ~H"""
+    <.subway_icon class={"bg-blue-500 #{@class}"}>BL</.subway_icon>
+    """
+  end
+
+  def mode_icon(%{type: :green_line_b} = assigns) do
+    ~H"""
+    <.subway_icon class={"bg-green-500 #{@class}"}>B</.subway_icon>
+    """
+  end
+
+  def mode_icon(%{type: :green_line_c} = assigns) do
+    ~H"""
+    <.subway_icon class={"bg-green-500 #{@class}"}>C</.subway_icon>
+    """
+  end
+
+  def mode_icon(%{type: :green_line_d} = assigns) do
+    ~H"""
+    <.subway_icon class={"bg-green-500 #{@class}"}>D</.subway_icon>
+    """
+  end
+
+  def mode_icon(%{type: :green_line_e} = assigns) do
+    ~H"""
+    <.subway_icon class={"bg-green-500 #{@class}"}>E</.subway_icon>
+    """
+  end
+
+  def mode_icon(%{type: :orange_line} = assigns) do
+    ~H"""
+    <.subway_icon class={"bg-orange-500 #{@class}"}>OL</.subway_icon>
+    """
+  end
+
+  def mode_icon(%{type: :red_line} = assigns) do
+    ~H"""
+    <.subway_icon class={"bg-red-500 #{@class}"}>RL</.subway_icon>
+    """
+  end
+
+  def mode_icon(%{type: :mattapan_line} = assigns) do
+    ~H"""
+    <.subway_icon class={"bg-red-500 #{@class}"}>M</.subway_icon>
+    """
+  end
+
+  def mode_icon(%{route_name: "BB", type: :logan_express} = assigns) do
+    ~H"""
+    <.logan_express_bb_icon class={"h-6 w-6 #{@class}"} />
+    """
+  end
+
+  def mode_icon(assigns) do
+    ~H"""
+    <.subway_icon class={"bg-silver-500 #{@class}"}>?</.subway_icon>
```

Comment:
`?` :sob:
I like the idea of a fallback icon but I don't think it should be labeled as a "subway" icon.